### DISCUSSION TITLE
grabber, a custom package manager

### DISCUSSION THREAD

**Discussion by redstone-dev** [AspectLinux/aspectlinux.github.io#4] [Ideas]
We could have our own package manager, complementary to `pacman`. I liked the name `bundle` a lot, but that's the name of Ruby's pm. So I changed it to `grabber`. What do you guys think?

Maybe it could also download and install the files necessary for the Aspects system.

**Comment by HGT-123** [comment]
I got an idea, what about the app project?
**A**spect
**P**ackage
**P**project

**Reply by redstone-dev** [reply]
Nah, I think `bundle` is a catchier name.

**Comment by michaeleldar** [comment]
Good idea!

The aspect/bundle system we are gonna use could easily be adapted to work as a package manager too!

**Reply by redstone-dev** [reply]
exactly why I came up with it :D

**Comment by bredcat999** [comment]
Sounds good! What will this be written in?

**Reply by redstone-dev** [reply]
Python for the moment, just to get it working, then we can rewrite it in a better (faster and more low-level) language.

**Reply by HGT-123** [reply]
I've alr started on Aspectscript as well

**Reply by HGT-123** [reply]
Ik - make a python eg:
```py
cmd = input(">>")
if cmd contains (bundle install):
  if cmd contains (Gnome)
      os.system("Sudo install gnome")
```
<br>

I know this is not proper python code but that is a general idea

**Reply by redstone-dev** [reply]
> I've alr started on Aspectscript as well

when you have an idea, please use the forum or the Ideas category on the org discussions to see if everyone thinks its a good idea.

**Comment by ajskateboarder** [comment]
Bundle is the name of Ruby's package manager. You may want to choose an even cooler name to prevent conflict :)

**Reply by redstone-dev** [reply]
okay... how about `supm`? short for "super package manager"

**Reply by HGT-123** [reply]
`supm install AspectPY` <br>
Its not bad...

**Reply by ajskateboarder** [reply]
What about simply `install`?  

`$ install some_random_package`

I know it's not catchy, but it totally makes sense

**Reply by redstone-dev** [reply]
hmmm... good idea.

**Reply by bredcat999** [reply]
would install have flags like `install --install firefox`

**Reply by ajskateboarder** [reply]
I'm sort of regretting `install` since you'll have to call it to remove packages (which is bad). Please scrap my idea :trollface:

**Reply by redstone-dev** [reply]
ok

**Reply by redstone-dev** [reply]
> I'm sort of regretting `install` since you'll have to call it to remove packages (which is bad). Please scrap my idea :trollface:

it's called `grabber` now

**Comment by bredcat999** [comment]
`install --uninstall install`

**Reply by redstone-dev** [reply]
no longer relevant as i changed it to `grabber`

**Comment by ajskateboarder** [comment]
Alright, thanks!

On Fri, Feb 3, 2023 at 9:39 PM RedstoneScratch ***@***.***>
wrote:

> I'm sort of regretting install since you'll have to call it to remove
> packages (which is bad). Please scrap my idea [image: :trollface:]
>
> it's called grabber now
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/orgs/AspectLinux/discussions/4#discussioncomment-4867491>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ATL54AEX4QJ4QLYCNJ5SRMLWVW6P5ANCNFSM6AAAAAATXN4GGM>
> .
> You are receiving this because you were mentioned.Message ID:
> <AspectLinux/aspectlinux.github.io/repo-discussions/4/comments/4867491@
> github.com>
>


-- 
-----------------------
I am Chikoo!

**Reply by HGT-123** [reply]
`sudo grabber install Aspectpy`
